src/figure/service.c

```diff
@@ -164,11 +201,33 @@ static void clinic_coverage(building *b)
     b->data.house.clinic = MAX_COVERAGE;
 }
 
+static void sickness_coverage(building *b, int coverage)
+{
+    int day = game_time_day();
+    // not sure because if day is the same but one month later or one year later it will be false
+    if (b->sickness_last_doctor_cure != day) {
```

Comment:
I didn't test how this works in-game, but is this check really needed?

Reply:
Yes, without this tricks, the doctor healing is too strong when he walks.